digdag-core/src/main/java/io/digdag/core/database/DatabaseMigrator.java

```diff
@@ -626,11 +626,117 @@ public void migrate(Handle handle)
         }
     };
 
+    private final Migration MigrateQueueRearchitecture = new Migration() {
+        @Override
+        public String getVersion()
+        {
+            return "20160719172538";
+        }
+
+        @Override
+        public void migrate(Handle handle)
+        {
+            // queued_task_locks
+            handle.update("drop table queued_task_locks");
+            handle.update("drop table queued_shared_task_locks");
+
+            handle.update(
+                    new CreateTableBuilder("queued_task_locks")
+                    .addLongId("id")  // references queued_tasks.id
+                    .addInt("site_id", "")
+                    .addInt("queue_id", "")
+                    .addInt("priority", "not null")
+                    .addInt("retry_count", "not null default 0")
+                    .addLong("lock_expire_time", "")
+                    .addString("lock_agent_id", "")
+                    .build());
+
+            handle.update("insert into queued_task_locks" +
+                    " (id, site_id, queue_id, priority)" +
+                    " select id, site_id, NULL, priority" +
+                    " from queued_tasks");
+
+            // queues
+            handle.update("alter table queues" +
+                    " add column shared_site_id int");
+
+            // resource_types
+            handle.update("drop table resource_types");
+
+            // queued_tasks
+            handle.update("alter table queued_tasks" +
+                    " alter column task_id drop not null");
+            handle.update("alter table queued_tasks" +
+                    " alter column queue_id drop not null");
+            handle.update("alter table queued_tasks" +
+                    " drop column priority");
+            handle.update("alter table queued_tasks" +
+                    " drop column resource_type_id");
+            handle.update("create unique index queued_tasks_on_site_id_task_id on queued_tasks (site_id, task_id)");
+
+            if (isPostgres()) {
+                handle.update(
+                    "CREATE FUNCTION lock_shared_tasks(target_site_id int, target_site_max_concurrency bigint, limit_count int, lock_expire_seconds int, agent_id text) returns setof bigint as $$\n" +
+                    "BEGIN\n" +
+                    "  IF pg_try_advisory_xact_lock(23300, target_site_id) THEN\n" +
```

Comment:
it's just a random number. PostgreSQL uses two INT numbers or one BIGINT number as the identifier of a lock